File: Estrutura_Repetição/Python/Ex1131.py
inter_vitorias = 0
gremio_vitorias = 0
empates = 0
grenais = 0

while True:
    gols_inter, gols_gremio = map(int, input().split())
    grenais += 1

    if gols_inter > gols_gremio:
        inter_vitorias += 1
    elif gols_gremio > gols_inter:
        gremio_vitorias += 1
    else:
        empates += 1

    resposta = int(input("Novo grenal (1-sim 2-nao)\n"))

    if resposta == 2:
        break

print(f"{grenais} grenais")
print(f"Inter:{inter_vitorias}")
print(f"Gremio:{gremio_vitorias}")
print(f"Empates:{empates}")

if inter_vitorias > gremio_vitorias:
    print("Inter venceu mais")
elif gremio_vitorias > inter_vitorias:
    print("Gremio venceu mais")
else:
    print("Nao houve vencedor")

File: Estrutura_Repetição/Clojure/Ex1131.clj
(ns Ex1131
  (:require [clojure.string :as str]))

(defn estatisticas-grenais [resultados]
  (let [total-grenais (count resultados)
        vitorias-inter (count (filter #(= % :inter) resultados))
        vitorias-gremio (count (filter #(= % :gremio) resultados))
        empates (count (filter #(= % :empate) resultados))
        vencedor (if (> vitorias-inter vitorias-gremio)
                   :inter
                   (if (> vitorias-gremio vitorias-inter)
                     :gremio
                     :empate))]
    (println total-grenais "grenais")
    (printf "Inter:%d\n" vitorias-inter)
    (printf "Gremio:%d\n" vitorias-gremio)
    (printf "Empates:%d\n" empates)
    (println (str/capitalize (name vencedor)) "venceu mais")))

(defn ler-resultado-grenal []
  (let [gols-inter (read)
        gols-gremio (read)]
    (if (> gols-inter gols-gremio)
      :inter
      (if (> gols-gremio gols-inter)
        :gremio
        :empate))))

(defn ler-resultados []
  (let [resultado (ler-resultado-grenal)]
    (println "Novo grenal (1-sim 2-nao)")
    (if (= (read) 1)
      (conj (ler-resultados) resultado)
      [resultado])))

(defn -main []
  (let [resultados (ler-resultados)]
    (estatisticas-grenais resultados)))

(-main)